def add_exception(self, future):
    with self.lock:
        super(_AsCompletedWaiter, self).add_exception(future)
        self.event.set()
